test/tests/functional/pbs_provisioning_enhancement.py

```diff
@@ -0,0 +1,310 @@
+                              max_attempts=20,
+                              interval=1)
+        self.assertTrue((nodes[0] == self.momA.shortname and
+                         nodes[1] == self.momB.shortname) or
+                        (nodes[0] == self.momB.shortname and
+                         nodes[1] == self.momA.shortname))
+
+    def test_subchunk_os_provisioning(self):
+        """
+        Test os provisioning job request consist of subchunks
+        with and without aoe resource.
+        """
+
+        a = {'Resource_List.select': '1:aoe=osimage1+1:ncpus=12'}
+        j = Job(TEST_USER1, a)
+        j.set_sleep_time(10)
+        jid = self.server.submit(j)
+        self.server.expect(JOB, ATTR_execvnode, id=jid, op=SET)
+        nodes = j.get_vnodes(j.exec_vnode)
+        self.assertTrue((nodes[0] == self.momA.shortname and
+                         nodes[1] == self.momB.shortname) or
+                        (nodes[0] == self.momB.shortname and
+                         nodes[1] == self.momA.shortname))
+        self.momA.restart()
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid)
+
+    def test_job_wide_provisioining_request(self):
+        """
+        Test jobs with jobwide aoe resource request.
+        """
+
+        # Below job will not run, since resource requested are job-wide,
+        # and no single node have all the requested resource.
+
+        j = Job(TEST_USER1)
+        j.set_sleep_time(5)
+        j.set_attributes({"Resource_List.aoe": "App1",
+                          "Resource_List.ncpus": 12})
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {ATTR_state: 'Q',
+                                 ATTR_comment:
+                                 (MATCH_RE, 'Not Running: Insufficient ' +
+                                  'amount of resource*')}, id=jid)
+
+        j = Job(TEST_USER1)
+        j.set_attributes({"Resource_List.aoe": "App1",
+                          "Resource_List.ncpus": 1})
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid)
+
+    def test_multiple_aoe_request(self):
+        """
+        Test jobs with multiple similar/various aoe request in subchunks.
+        Job request cosisting of multiple subchunks with different aoe will
+        fail submission, Whereas job request with same aoe across multiple
+        subchunks should be succesful.
+        """
+
+        a1 = {'Resource_List.select':
+              '1:ncpus=1:aoe=App1+1:ncpus=12:aoe=osimage1'}
+
+        a2 = {'Resource_List.select':
+              '1:ncpus=1:aoe=App1+1:ncpus=12:aoe=App1'}
+
+        # Below job will fail submission, since different aoe's requested,
+        # across multiple subchunks.
+
+        j = Job(TEST_USER1)
+        j.set_attributes(a1)
+        j.set_sleep_time(5)
+        try:
+            jid = self.server.submit(j)
+            self.assertTrue(jid is None, 'Job successfully submitted' +
```

Comment:
Set jid to None before the try/except block

Reply:
Done!